keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -2166,6 +2279,16 @@ k8s.io/kubelet/pkg/apis/dra gRPC interface. It was inspired by
 [CSI](https://github.com/container-storage-interface/spec/blob/master/spec.md),
 with “volume” replaced by “resource” and volume specific parts removed.
 
+Versions v1alpha4 and v1beta1 are supported by kubelet. Both are identical.
+DRA drivers should implement both because support for v1alpha4 might get
+removed.
+
+The following metric mirrors the `csi_operations_seconds`:
+- Metric name: `dra_operations_seconds`
```

Comment:
Would it make sense to name it so that it's clear that this metric is about GRPC, e.g. `dra_grpc_operations_seconds` or something similar?
We may decide to add other metrics for DRA operations, so it would be nice if they don't conflict.

Reply:
That's then not consistent with other metrics. Let's ask our SIG Instrumentation reviewers during the implementation review what they prefer.

Reply:
Well, even `dra_operations_seconds` is not consistent with similar metrics in the pkg/kubelet/metrics/metrics.go. We should probably decide which naming convention we're going to follow [csi](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/metrics.go) or [kubelet](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go). 
As we're going to add new metrics to the kubelet, I tend to prefer kubelet naming.

Reply:
We don't need to resolve that here, we can update the KEP based on what SIG instrumentation recommends during the implementation phase.

Reply:
So for Kubelet naming convention the metrics could be named this way:
- `dra_operations_duration_seconds`
- `dra_operations_errors_total`
- `dra_grpc_operations_duration_seconds`
- `dra_grpc_operations_errors_total`

And for CSI they can be named this way:
- `dra_operation_seconds`
- `operations_seconds`
similar to [storage_operation_duration_seconds](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/metrics.go#L48) and [operations_seconds](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/metrics.go#L66) existing CSI metrics. The latter can't be done as it would conflict with the existing metric name.

I'd propose to use Kubelet naming scheme as it's much less confusing then CSI from my point of view.

Reply:
I thought though that we didn't need the separate errors_total because we have it in the duration_seconds histograms? Or is that an aggregation (not sure we need that?)?

Reply:
> We don't need to resolve that here, we can update the KEP based on what SIG instrumentation recommends during the implementation phase.

absolutely

> I thought though that we didn't need the separate errors_total because we have it in the duration_seconds histograms? Or is that an aggregation (not sure we need that?)?

That's what kubelet uses for its metrics, e.g. [runtime_operations_duration_seconds](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L381) and [runtime_operations_errors_total](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/metrics/metrics.go#L393)

Reply:
> I'd propose to use Kubelet naming scheme as it's much less confusing then CSI from my point of view.

`csi_operations_seconds` is also a kubelet metric.

Let's continue the discussion in your PR or in #wg-device-management. You can present the alternatives and perhaps some future users of these metrics will have an opinion.

Reply:
ok, let's continue there